include/sycl_ext_complex.hpp

```diff
@@ -244,14 +243,27 @@ template<class T> complex<T> tanh (const complex<T>&);
 
 // clang-format on
 
-#define _SYCL_EXT_CPLX_BEGIN_NAMESPACE_STD namespace sycl::ext::cplx {
+#ifndef _SYCL_CPLX_NAMESPACE
+#ifdef __HIPSYCL__
+#define _SYCL_CPLX_NAMESPACE hipsycl::sycl::ext::cplx
+#else
+#define _SYCL_CPLX_NAMESPACE sycl::ext::cplx
+#endif
+#endif
+
+#define _SYCL_EXT_CPLX_BEGIN_NAMESPACE_STD namespace _SYCL_CPLX_NAMESPACE {
 #define _SYCL_EXT_CPLX_END_NAMESPACE_STD }
 #define _SYCL_EXT_CPLX_INLINE_VISIBILITY                                       \
-  inline __attribute__((__visibility__("hidden"), __always_inline__))
+  [[gnu::always_inline]] [[clang::always_inline]] inline
 
 #include <complex>
 #include <sstream> // for std::basic_ostringstream
+#if (__INTEL_CLANG_COMPILER && __INTEL_CLANG_COMPILER < 20230000)
```

Comment:
These MACROs create the same issue seen in that [PR](https://github.com/argonne-lcf/SyclCPLX/pull/18)

We can  use ```SYCL_IMPLEMENTATION_ONEAPI``` and ```__SYCL_COMPILER_VERSION```
Otherwise, what about ```__has_include``` ?

Reply:
This is not about determining Intel vs other implemntation - it's about checking if we are using a oneAPI release that uses `CL/sycl.hpp` instead of `sycl/sycl.hpp`. Based on looking at list of all macros defined by different compilers, `__INTEL_CLANG_COMPILER` is the only one I found that can be used to make this distinction.

The open source intel/llvm fork does not have this variable, but it also uses the new `sycl/sycl.hpp` header, so it's fine, it will fallback to the other branch and works.

Reply:
Oh I wasn't aware of `__has_include`, standard in 17, maybe that is a cleaner option. I will test it out.

Reply:
Just add another `elif` : `No sycl header found` :)

Reply:
Force pushed change to use has_include instead, with #error if neither found.